File: src/quest_editor/actions/EditNameAction.ts
import { QuestEditAction } from "./QuestEditAction";

export class EditNameAction extends QuestEditAction<string> {
    readonly description = "Edit name";

    undo(): void {
        this.quest.set_name(this.old);
    }

    redo(): void {
        this.quest.set_name(this.new);
    }
}

File: web/src/main/kotlin/world/phantasmal/web/questEditor/actions/EditNameAction.kt
package world.phantasmal.web.questEditor.actions

import world.phantasmal.web.core.actions.Action
import world.phantasmal.web.questEditor.models.QuestModel

class EditNameAction(
    private val quest: QuestModel,
    private val newName: String,
    private val oldName: String,
) : Action {
    override val description: String = "Edit name"

    override fun execute() {
        quest.setName(newName)
    }

    override fun undo() {
        quest.setName(oldName)
    }
}
